clickplc/driver.py

```diff
@@ -385,15 +446,16 @@ async def _get_dd(self, start: int, end: int | None) -> dict | int:
         address = 16384 + 2 * (start - 1)
         count = 2 if end is None else 2 * (end - start + 1)
         registers = await self.read_registers(address, count)
-
-        # Pack registers as 16-bit unsigned shorts, little-endian ('<'), then unpack as signed 32-bit ints
-        packed = struct.pack(f'<{count}H', *registers)
-        values = struct.unpack(f'<{count // 2}i', packed)  # 'i' = signed 32-bit int
-
-        if count == 2:
-            return values[0]
-        return {f'dd{start + i}': v for i, v in enumerate(values)}
-
+        registers_real = registers.copy()
```

Comment:
what's going on here - does the existing code not work?  (it's certainly possible I introduced a bug...)